src/App.tsx

```diff
@@ -15,36 +16,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  None = '',
+}
+
 export const App: React.FC = () => {
+  const [items, setItems] = useState<string[]>(goodsFromServer);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [initialItems] = useState<string[]>(goodsFromServer);
+
+  function sortByAlphabet() {
+    if (sortType === SortType.Alphabet && !isReversed) {
+      return;
+    }
+
+    const sortedItems = [...items].sort((a, b) => a.localeCompare(b));
+
+    if (isReversed) {
+      sortedItems.reverse();
+    }
+
+    setItems(sortedItems);
+    setSortType(SortType.Alphabet);
+  }
+
+  function sortByLength() {
+    if (sortType === SortType.Length && !isReversed) {
+      return;
+    }
+
+    const sortedItems = [...items].sort((good1, good2) => {
+      if (good1.length === good2.length) {
+        return good1.localeCompare(good2);
+      }
+
+      return good1.length - good2.length;
+    });
+
+    if (isReversed) {
+      sortedItems.reverse();
+    }
+
+    setItems(sortedItems);
+    setSortType(SortType.Length);
+  }
+
+  const reset = () => {
+    setItems(initialItems);
+    setSortType(SortType.None);
+    setIsReversed(false);
+  };
```

Comment:
The function name 'reset' is not descriptive enough and does not start with a verb. Consider renaming it to something like 'handleReset' to make the function's purpose clearer and to comply with the checklist.